screens/Authentication/SignUp/CreateCredentials.tsx

```diff
@@ -0,0 +1,186 @@
+  }, [isError, isInputValid, isLoading])
+
+  const badgeName: { [key: number]: string } = {
+    1: "3/4",
+    2: "4/4",
+  }
+  const titleText: { [key: number]: string } = {
+    1: "What is your name?",
+    2: "Add your username",
+  }
+  const inputPlaceholder: { [key: number]: string } = {
+    1: "Full name",
+    2: "Username",
+  }
+  const descriptionText: { [key: number]: string } = {
+    1: "Add your name so friends can find you.",
+    2: "Usernames can only contain letter, numbers, underscores and periods.",
+  }
+
+  const buttonText = useMemo(() => {
+    if (isLoading || isError) return "" // we then use the <ButtonChildren component
+
+    return "NEXT"
+  }, [isError, isLoading])
+
+  const handleUsernameSubmit = async () => {
+    setIsLoading(true)
+    // setTimeout to mock fetching username from the backend.
+    await new Promise((resolve) => setTimeout(resolve, 2000))
+    if (text === "Username123") setIsError(true)
+    setIsLoading(false)
+  }
+
+  const handleNameSubmit = () => {
+    // TODO: logic for name submission.
+    setText("")
+    setStep(2)
+  }
+
+  const handleSubmit = async () => {
+    if (!isInputValid) return
+
+    if (step === 1) handleNameSubmit()
+    else await handleUsernameSubmit()
+  }
+
+  // Clear the states on every keystroke
+  useEffect(() => {
+    setIsLoading(false)
+    setIsError(false)
+  }, [text])
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.navigation}>
+        <View
+          style={{
+            flexGrow: 1,
+          }}>
+          <Badge
+            variant="primary"
+            name={badgeName[step]}
+            style={[isDisabled && styles.disabledBadge]}
+          />
+        </View>
+        <TouchableOpacity
+          style={styles.crossMarkContainer}
+          onPress={() => navigation.push("Home")}>
+          <CrossMark color={theme.colors.white} />
+        </TouchableOpacity>
+      </View>
+      <View style={styles.mainContent}>
+        <CustomText style={styles.title} size="lg">
+          {titleText[step]}
+        </CustomText>
+        <Input
+          handleTextChange={setText}
+          variant="secondary"
+          placeholder={inputPlaceholder[step]}
+          value={text}
+          onSubmitEditing={handleSubmit}
+          autoFocus
+        />
+        <CustomText size="base" style={styles.description}>
+          {descriptionText[step]}
+        </CustomText>
+      </View>
+      <CustomButton
+        disabled={isDisabled}
+        fullWidth
+        title={buttonText}
+        variant="primary"
+        containerStyles={{ width: "100%" }}
+        textStyles={styles.buttonText}
+        onPress={handleSubmit}>
+        {(isLoading || isError) && (
+          <ButtonChildren
+            text={
+              isLoading ? "checking availability" : "username not available"
+            }
+            icon={isError ? "info" : "loader"}
+          />
+        )}
+      </CustomButton>
```

Comment:
can we rename the `ButtonChildren` something like "Credential button" or another name that make sense to you, and the have thus whole button in the component. Am suggesting this because having a component called `ButtonChildren`feels a bit odd and doesn't clearly describe its functionality